### PR TITLE
CLOUDP-116175: logs decrypt with KMIP e2e test

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes

_Jira ticket:_ CLOUDP-116175

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: build/ci/kmip/start.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
+import copy
```
Comment (tibulca) [inline build/ci/kmip/start.py:1]: script copied from server tests. Only ther main function was updated

File: e2e/cloud_manager/decryption_kmip/tls-localhost.pem [hunk 2]
```diff
@@ -0,0 +1,52 @@
+-----BEGIN CERTIFICATE-----
```
Comment (tibulca) [inline e2e/cloud_manager/decryption_kmip/tls-localhost.pem:1]: replacing with evergreen env vars

File: build/ci/evergreen.yml [hunk 3]
```diff
@@ -282,6 +284,20 @@ functions:
...
+        add_to_path:
+          - ${go_bin}
+          - ../../bin
+        working_dir: /src/github.com/mongodb/mongocli/build/ci/kmip
+        binary: ./install.sh
+        args: ["-h", "hosts.json"]
+        env:
+          XDG_CONFIG_HOME: ${workdir}
+          KMIP_CA: ${logs_decrypt_kmip_ca}
+          KMIP_CERT: ${logs_decrypt_kmip_cert}
```
Comment (tibulca) [inline build/ci/evergreen.yml:300]: the kmip server (same as the cli) needs the same CA, cert and db (where the key encryption keys are stored) to be able to decrypt existing logs so generating new certs won't work.

File: e2e/cloud_manager/decryption_kmip/test1-input [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{"ts":{"$date":{"$numberLong":"1648640636381"}},"version":"0.0","compressionMode":"zstd","keyStoreIdentifier":{"provider":"kmip","uid":"2","kmipServerName":["localhost"],"kmipPort":{"$numberInt":"8081"},"keyWrapMethod":"get"},"encryptedKey":{"$binary":{"base64":"UAAAAAVpdgAMAAAAAOuTE5U6tgzWdiEw2AVrZXkALAAAAADKurLHG5ED3swCFM3SNwHUjyDjUx5RollQdH5KcManix72zerd4Fjcaj4mTAA=","subType":"0"}},"MAC":"ktXd/Z159KMSq6heAQAAAAAAAAAAAAAA","auditRecordType":"header"}
```
Comment (tibulca) [inline e2e/cloud_manager/decryption_kmip/test1-input:1]: this is using KMIP GET method (see on this line `"keyWrapMethod":"get"`)

File: e2e/cloud_manager/decryption_kmip/test2-input [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{"ts":{"$date":{"$numberLong":"1648685783241"}},"version":"0.0","compressionMode":"zstd","keyStoreIdentifier":{"provider":"kmip","uid":"1","kmipServerName":["localhost"],"kmipPort":{"$numberInt":"8081"},"keyWrapMethod":"encrypt"},"encryptedKey":{"$binary":{"base64":"WAAAAAVpdgAQAAAAAFSuxGWGC88Yi0dVi4QQTmYFa2V5ADAAAAAAuKH7TQXBSfyA/Oy9EGSnjkZ8+ooRlevbg6rfN34h/KNAN1fEo/eOb5tufQTItgcRAA==","subType":"0"}},"MAC":"JtRlaXS91rD79M4ZAQAAAAAAAAAAAAAA","auditRecordType":"header"}
```
Comment (tibulca) [inline e2e/cloud_manager/decryption_kmip/test2-input:1]: this is using KMIP Encrypt wrap method (see on this line "keyWrapMethod":"encrypt")

**Comment by johansteyn** [comment]
The Go code LGTM.

File: e2e/cloud_manager/decryption_kmip/test1-output [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{"atype":"startup","uuid":{"$binary":{"base64":"skhKjh22RM2vbNaHcSafEQ==","subType":"04"}},"users":[],"roles":[],"result":0,"ts":{"$date":"2022-03-30T11:43:56.383Z"},"local":{"isSystemUser":true},"remote":{"isSystemUser":true},"param":{"options":{"security":{"kmip":{"serverCAFile":"/root/pykmip/tls-rootCA.pem","serverName":"localhost","clientCertificateFile":"/root/pykmip/tls-localhost.pem","port":8081}},"setParameter":{"auditEncryptKeyWithKMIPGet":"true"},"auditLog":{"destination":"file","format":"JSON","path":"auditLogKmipGet.json","auditEncryptionKeyIdentifier":"2","compressionMode":"zstd"}}}}
```
[the remark this answers is not in the document]
Comment (tibulca) [reply e2e/cloud_manager/decryption_kmip/test1-output]: e2e/cloud_manager/decryption/* is obviously the better option, thanks ;)

**Review by fmenezes — APPROVED** [review]
Thanks for the changes

**Pull request merged**